Generator/Generator/BuiltinGen.swift

```diff
@@ -554,33 +554,26 @@ func generateBuiltinClasses (values: [JGodotBuiltinClass], outputDir: String?) a
                     }
                 }
             }
+            let gtype = gtypeFromTypeName (bc.name)
+            
+            p("public static var gType: Variant.GType { .\(gtype) }")
             
             p ("/// Creates a new instance from the given variant if it contains a \(typeName)")
-            let gtype = gtypeFromTypeName (bc.name)
             // Now generate the variant constructor
             if kind == .isClass {
-                p ("public required init? (_ from: Variant)") {
-                    p ("guard from.gtype == .\(gtype) else") {
-                        p ("return nil")
-                    }
+                p("public static func unwrap(variant: Variant) -> \(typeName)?") {
+                    p("guard variant.gtype == .\(gtype) else { return nil }")
                     p ("var localContent: \(typeName).ContentType = \(typeName).zero")
-                    p ("from.toType(.\(gtype), dest: &localContent)")
-                    p ("// Replicate the constructor, because of a lame Swift requirement")
-                    p ("var args: [UnsafeRawPointer?] = []")
-                    p ("withUnsafePointer (to: &localContent)", arg: " ptr in") {
-                        p ("args.append (ptr)")
-                        p ("\(typeName).constructor1 (&content, &args)")
-                    }
+                    p ("variant.toType(.\(gtype), dest: &localContent)")
+                    p ("return \(typeName)(content: localContent)")
                 }
             } else {
-                p ("public init? (_ from: Variant)") {
-                    p ("guard from.gtype == .\(gtype) else") {
-                        p ("return nil")
-                    }
-                    p ("var v = \(bc.name)()")
-                    p ("from.toType(.\(gtype), dest: &v)")
-                    p ("self.init (from: v)")
-                }                
+                p("public static func unwrap(variant: Variant) -> Self?") {
+                    p("guard variant.gtype == .\(gtype) else { return nil }")
+                    p("var value = \(bc.name)()")
+                    p("variant.toType(.\(gtype), dest: &value)")
+                    p("return value")
+                }
```

Comment:
Perhaps this might be me, but I feel that the initializer was more appropriate in this context, and not specifically for type casting reasons. Rather, because we are constructing something with that variant. `unwrap` would imply to me that Variant is actually a representation of an optional, which, I think, might be a little confusing.